streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/DruidEventMapper.java

```diff
@@ -37,7 +39,60 @@ public DruidEventMapper(String eventFiledName) {
 
     @Override
     public Map<String, Object> getEvent(ITuple tuple) {
+
         StreamlineEvent event = (StreamlineEvent) tuple.getValueByField(eventFiledName);
-        return event.addFieldAndValue(DruidBeamFactoryImpl.PROCESSING_TIME, System.currentTimeMillis());
+        return new DruidEvent(event.addFieldAndValue(DruidBeamFactoryImpl.PROCESSING_TIME, System.currentTimeMillis()));
+    }
+
+    public static class DruidEvent extends ForwardingMap<String, Object> {
+
+        private final StreamlineEvent event;
+
+        public DruidEvent(StreamlineEvent event) {
+            this.event = event;
+        }
+
+        public static boolean isMap(Object map) {
+            return map instanceof Map;
+        }
+
+        @Override
+        protected Map<String, Object> delegate() {
+            return event;
+        }
+
+        @Override
+        public Object get(@Nullable Object key) {
+            Object value = super.get(key);
```

Comment:
Can you see if you can replace code in this method using https://github.com/hortonworks/streamline/blob/master/streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/StreamlineRuntimeUtil.java#L24
That way we can have everything in one place.